parlai/tasks/triviaqa/agents.py

```diff
@@ -123,13 +133,42 @@ def __init__(self, opt, shared=None):
         super().__init__(opt, shared)
 
 
+class NoEvidenceWikipediaTeacher(WikipediaTeacher):
+    def __init__(self, opt, shared=None):
+        self.no_evidence = True
+        super().__init__(opt, shared)
+
+
 class VerifiedTeacher(MultiTaskTeacher):
     def __init__(self, opt, shared=None):
         opt = copy.deepcopy(opt)
         opt['task'] = 'triviaqa:VerifiedWikipedia,triviaqa:VerifiedWeb'
         super().__init__(opt, shared)
 
 
+class NoEvidenceUnionTeacher(DialogTeacher):
+    def __init__(self, opt, shared=None):
+        if not hasattr(self, 'prefix'):
+            self.prefix = ''
+            self.suffix = 'train' if opt['datatype'].startswith('train') else 'dev'
+
+        qa_dir, self.evidence_dir = _path(opt)
+        opt['datafile'] = os.path.join(
+            qa_dir, self.prefix + 'union-noevidence-' + self.suffix + '.json'
+        )
+        self.id = 'triviaqa'
```

Comment:
can we make the id like `Trivia QA No Evidence Union`

Reply:
adding triviaqa feels weird cause the other files don't have it, but yeah, should be consistent in order 🙃